addons/partner_autocomplete/static/src/js/partner_autocomplete_many2one.js

```diff
@@ -0,0 +1,86 @@
+odoo.define('partner.autocomplete.many2one', function (require) {
+    'use strict';
+
+    var PartnerFieldMany2One = require('web.res_partner_many2one').PartnerFieldMany2One;
+    var core = require('web.core');
+    var Autocomplete = require('partner.autocomplete.core');
+
+    var _t = core._t;
+
+    var PartnerField = PartnerFieldMany2One.include({
+        /**
+         * @override
+         */
+        init: function () {
+            this._super.apply(this, arguments);
+            this._addAutocompleteSource(this._searchSuggestions, 'Searching Autocomplete...', 20, function(term){
+                return Autocomplete.validateSearchTerm(term)
+            });
+        },
+
+        /**
+         * Query Autocomplete and add results to the popup
+         *
+         * @override
+         * @param search_val
+         * @returns {*}
```

Comment:
Variable API is bad practice, what about making this function always return a Deferred resolved with an Array, which is sometimes empty?